### ISSUE TITLE
Hermite time integration is unconditionally unstable (missing integrating factor for phase mixing)

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#120]
## Summary

The Hermite moment hierarchy (`g`) uses plain RK2 for time integration, but the linear phase-mixing terms are purely oscillatory (imaginary eigenvalues). The midpoint RK2 stability function satisfies `|R(iω)|² = 1 + ω⁴/4 > 1` for all `ω ≠ 0`, making it **unconditionally unstable** for these terms. Any nonzero `g` grows exponentially regardless of timestep or collisional damping.

This contrasts with z±, which correctly uses integrating factors (`exp(±ikz·dt)`) to handle their oscillatory linear terms analytically.

## Reproduction

```python
from krmhd.config import SimulationConfig
from krmhd.physics import initialize_hermite_moments, KRMHDState

config = SimulationConfig.from_yaml("any_config_with_M_gt_0.yaml")
grid = config.create_grid()
state = config.create_initial_state(grid)

# Seed g with tiny perturbation
g_seed = initialize_hermite_moments(grid, 32, perturbation_amplitude=1e-6, seed=42)
state = KRMHDState(
    z_plus=state.z_plus, z_minus=state.z_minus,
    B_parallel=state.B_parallel, g=g_seed,
    M=state.M, beta_i=state.beta_i, v_th=state.v_th,
    nu=state.nu, Lambda=state.Lambda, time=state.time, grid=state.grid,
)

# g blows up within ~50 steps regardless of nu, eta, or dt
```

## Root cause

In `timestepping.py`, z± uses integrating factors (lines 377-400):
```python
phase_plus_half = exp(+1j * kz * dt / 2)
z_plus_half = phase_plus_half * (z_plus + phase_plus_half * dt/2 * NL)
```

But g uses plain RK2 (lines 402-434):
```python
g_half = fields.g + (dt / 2) * rhs_0.g       # no integrating factor
g_new  = fields.g + dt * rhs_half.g
```

The linear phase-mixing terms in `gm_rhs` — `sqrt(beta) · ikz · [sqrt((m+1)/2)·g_{m+1} + sqrt(m/2)·g_{m-1}]` — are purely oscillatory and need the same treatment as the `±ikz` terms in the Elsasser equations.

## Growth rate estimate

The Hermite tridiagonal coupling matrix T has T_{m,m+1} = sqrt((m+1)/2), T_{m,m-1} = sqrt(m/2). Its maximum eigenvalue is ~sqrt(2M). The maximum oscillation frequency is:

```
ω_max = sqrt(beta) · kz_max · sqrt(2M)
```

For M=32, kz_max=4, beta=1: ω_max ≈ 32.  
With dt=0.118 (dev grid CFL): ω·dt ≈ 3.8, giving |R| ≈ 7x amplification per step.  
After 50 steps: 7⁵⁰ ≈ 10⁴², matching observed blowup.

### Quick stopgap: Hermite CFL constraint

Add the phase-mixing eigenvalue to `compute_cfl_timestep`:

```python
omega_hermite = jnp.sqrt(2 * M) * jnp.max(jnp.abs(kz)) * jnp.sqrt(beta_i)
dt_hermite = 0.1 / omega_hermite  # keep omega*dt ≈ 0.1
dt = min(dt_cfl, dt_hermite)
```

This makes the instability slow enough (~0.0025% per step) for collisional damping to dominate. Costs ~2-4x more steps but requires minimal code changes.

### Proper fix: Integrating factors for the Hermite hierarchy

The linear operator `dg/dt = -sqrt(beta)·ikz·T·g` has a known matrix structure. T is real symmetric, so its eigendecomposition `T = U·D·Uᵀ` can be precomputed once at initialization.

Per step, apply the exact propagator:
```python
# Precomputed once at init:
eigenvalues, U = np.linalg.eigh(T)  # (M+1) x (M+1)

# Per step, per kz (half-step Strang splitting):
phase = exp(-sqrt(beta) * 1j * kz * eigenvalues * dt/2)
g_rotated = U.T @ g          # O(M²) per k-point
g_rotated *= phase            # O(M)
g = U @ g_rotated             # O(M²)
# ... RK2 for nonlinear terms only ...
# ... second half-step ...
```

This mirrors the z± approach: oscillatory linear part solved exactly, RK2 handles only (CFL-stable) nonlinear advection, exponential step handles collisions.

## Impact

All kinetic simulations (M > 0 with nonzero g) are affected. This blocks Study 02 (collisionality scan) and any work requiring the Hermite cascade.

## Also note

The collision operator in `gm_rhs` (line 1122) uses `nu * m * gm` (unnormalized), while the exponential step in `gandalf_step` (line 476) uses `nu * (m/M)^hyper_n` (normalized). This appears to double-count collisions with different normalizations. Might be intentional but worth verifying.
